lib/MusicBrainz/Server/Controller/Role/Merge.pm

```diff
@@ -195,22 +206,30 @@ role {
         log_assertion { @old_ids >= 1 } 'Got at least 1 entity to merge';
 
         $c->model('MB')->with_transaction(sub {
-            $self->_insert_edit(
-                $c, $form,
-                edit_type => $params->edit_type,
-                new_entity => {
-                    id => $new->id,
-                    name => $new->name,
-                    $self->_extra_entity_data($c, $form, $new)
-                },
-                old_entities => [ map +{
-                    id => $entity_id{$_}->id,
-                    name => $entity_id{$_}->name,
-                    $self->_extra_entity_data($c, $form, $entity_id{$_})
-                }, @old_ids ],
-                (map { $_->name => $_->value } $form->edit_fields),
-                $self->_merge_parameters($c, $form, $entities)
-            );
+            if ($params->edit_type) {
+                $self->_insert_edit(
+                    $c, $form,
+                    edit_type => $params->edit_type,
+                    new_entity => {
+                        id => $new->id,
+                        name => $new->name,
+                        $self->_extra_entity_data($c, $form, $new)
+                    },
+                    old_entities => [ map +{
+                        id => $entity_id{$_}->id,
+                        name => $entity_id{$_}->name,
+                        $self->_extra_entity_data($c, $form, $entity_id{$_})
+                    }, @old_ids ],
+                    (map { $_->name => $_->value } $form->edit_fields),
+                    $self->_merge_parameters($c, $form, $entities)
+                );
+            } else {
+                $c->model('Collection')->merge(
```

Comment:
I think it's a good idea to check if the entity type is 'collection' here (even if it's the only thing for which `$params->edit_type` is unset right now).

Reply:
I think the only way we have to do this is by checking `@$entities[0]->entity_type` - am I missing a better option? Made that change for now.

Reply:
I think you can check `$c->namespace eq 'collection` too. Or what you have, but should be `$entities->[0]->entity_type` if it's an array ref.